gprofiler/profilers/java.py

```diff
@@ -259,11 +285,109 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
+        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid, "jvm_flags": self.get_jvm_flags(process)}
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    @functools.lru_cache(maxsize=1024)
+    def get_jvm_flags(self, process: Process) -> Optional[List[tuple]]:
+        jvm_raw_flags = self.get_jvm_flags_raw(process)
+        if jvm_raw_flags is None:
+            return None
+
+        return [dataclasses.astuple(flag) for flag in jvm_raw_flags if self.filter_jvm_flags(flag)]
+
+    @staticmethod
+    def filter_jvm_flags(flag: JvmFlag) -> bool:
+        if flag.origin == "default":
+            return False
+
+        if flag.type not in [
+            "bool",
+            "int",
+            "uint",
+            "intx",
+            "uintx",
+            "uint64_t",
+            "size_t",
+            "double",
+        ]:
+            return False
+
+        if set(flag.kind).intersection({"notproduct", "pd", "develop"}):
+            return False
+
+        return True
+
+    def get_jvm_flags_raw(self, process: Process) -> Optional[List[JvmFlag]]:
+        """
+        The output of VM.flags -all format on jdk 8:
+        bool UseCompressedClassPointers               := true                                {lp64_product}
+        flag_type flag_name                           := flag_value                          {flag_kind}
+        ":=" indicates non default origin for the flag, while "=" indicates default origin
+
+        The output of VM.flags -all format on jdk 9+:
+        bool OptoScheduling                           = false                               {C2 pd product} {default}
+        flag_type flag_name                           = flag_value                          {flag_kind} {flag_origin}
+
+        flag_kind is space separated list of kinds, e.g. "C2 pd product"
+
+        possible flag kinds: "product", "manageable", "diagnostic", "experimental", "notproduct", "develop",
+        "lp64_product", "rw", "pd", "JVMCI", "C1", "C2", "ARCH"
+
+        possible flag types: "bool", "int", "uint", "intx", "uintx", "uint64_t", "size_t", "double", "ccstr",
+        "ccstrlist"
+
+        possible flag origins: default, non-default, command line, environment, config file, management, ergonomic,
+        attach, internal, jimage
+        """
+
+        try:
+            output = self.jattach_jcmd_runner.run(process, "VM.flags -all")
+        except CalledProcessError as e:
+            logging.warning(f"Couldn't get jvm flags for process {process.pid}: {e.stderr}")
```

Comment:
```suggestion
            logging.warning("Couldn't get jvm flags for process", pid=process.pid, stderr=e.stderr)
```
Prefer extras to formatting into messages

Reply:
@uriariel Please note that you used `%s` formatting instead of extra args like suggested here.

Reply:
extra args should be defined in the logger format, don't they?

Reply:
@Jongy Can you clarify? AFAICT any extra argument other than some built-ins which have special meaning (like `exc_info`) are fine.

Reply:
@YoniKF - the convention in gProfiler is to pass extras this way. Non-logging kwargs are converted to extras - it is handled by `ExtraAdapter` in glogger. Please follow it and pass extras that way (and refrain from using `%` formatting anyway lol, resort to `.format()` but not `%` please)